etl/src/replication/apply.rs

```diff
@@ -552,9 +552,9 @@ where
 
             // PRIORITY 3: Handle table synchronization coordination signals.
             // Table sync workers signal when they complete initial data copying and are ready
-            // to transition to continuous replication mode. We use map_or_else with pending()
-            // to make this branch optional - if no signal receiver exists, this branch never fires.
-            _ = force_syncing_tables_rx.as_mut().map_or_else(|| pending().boxed(), |rx| rx.changed().boxed()) => {
+            // to transition to continuous replication mode. Guard the branch so it stays
+            // dormant if no signal receiver was provided.
+            _ = force_syncing_tables_rx.as_mut().unwrap().changed(), if force_syncing_tables_rx.is_some() => {
```

Comment:
Instead of having the ugly boxing, we instead use an `unwrap` without a compile time guarantee but rather just the conditional on the branch.